From 3d37af699dd1e0a27b6b3e4f2fef7e673574564a Mon Sep 17 00:00:00 2001
From: Szabolcs Nagy <szabolcs.nagy@arm.com>
Date: Thu, 9 Nov 2023 13:35:37 +0000
Subject: [PATCH 171/180] ld: aarch64: Use lp64 abi in recent BTI stub tests

The tests are not compatible with ilp32 abi: the GNU property
note is ABI dependent (size changes) and the disasm is ABI
dependent too.  Making the test portable between the ABIs is
not trivial.

For now force lp64 abi.

(cherry picked from commit 7b0c124970d0dba1703284189f09be2cbe17fa91)
---
 ld/testsuite/ld-aarch64/aarch64-elf.exp | 8 ++++----
 ld/testsuite/ld-aarch64/bti-far-1.d     | 1 -
 ld/testsuite/ld-aarch64/bti-far-2.d     | 1 -
 ld/testsuite/ld-aarch64/bti-far-3.d     | 1 -
 ld/testsuite/ld-aarch64/bti-far-opt.d   | 1 -
 5 files changed, 4 insertions(+), 8 deletions(-)

diff --git a/ld/testsuite/ld-aarch64/aarch64-elf.exp b/ld/testsuite/ld-aarch64/aarch64-elf.exp
index fa5ac3a2cdd..37e3b82af57 100644
--- a/ld/testsuite/ld-aarch64/aarch64-elf.exp
+++ b/ld/testsuite/ld-aarch64/aarch64-elf.exp
@@ -463,10 +463,10 @@ run_dump_test "bti-warn"
 run_dump_test "weak-tls"
 run_dump_test "undef-tls"
 
-run_dump_test "bti-far-1"
-run_dump_test "bti-far-2"
-run_dump_test "bti-far-opt"
-run_dump_test "bti-far-3"
+run_dump_test_lp64 "bti-far-1"
+run_dump_test_lp64 "bti-far-2"
+run_dump_test_lp64 "bti-far-opt"
+run_dump_test_lp64 "bti-far-3"
 
 if { ![skip_sframe_tests] } {
   run_dump_test "sframe-simple-1"
diff --git a/ld/testsuite/ld-aarch64/bti-far-1.d b/ld/testsuite/ld-aarch64/bti-far-1.d
index d2dbc9db110..3ca3f1654bd 100644
--- a/ld/testsuite/ld-aarch64/bti-far-1.d
+++ b/ld/testsuite/ld-aarch64/bti-far-1.d
@@ -1,7 +1,6 @@
 #name: Check linker stubs with indirect calls handle BTI (shared lib).
 #source: bti-far.s
 #target: [check_shared_lib_support]
-#as: -mabi=lp64
 #ld: -shared -T bti-far.ld
 #objdump: -dr
 
diff --git a/ld/testsuite/ld-aarch64/bti-far-2.d b/ld/testsuite/ld-aarch64/bti-far-2.d
index b859e6ce399..ec26e368900 100644
--- a/ld/testsuite/ld-aarch64/bti-far-2.d
+++ b/ld/testsuite/ld-aarch64/bti-far-2.d
@@ -1,6 +1,5 @@
 #name: Check linker stubs with indirect calls handle BTI (exe).
 #source: bti-far.s
-#as: -mabi=lp64
 #ld: -T bti-far.ld
 #objdump: -dr
 
diff --git a/ld/testsuite/ld-aarch64/bti-far-3.d b/ld/testsuite/ld-aarch64/bti-far-3.d
index 1410b1389b2..b27d8b56548 100644
--- a/ld/testsuite/ld-aarch64/bti-far-3.d
+++ b/ld/testsuite/ld-aarch64/bti-far-3.d
@@ -2,7 +2,6 @@
 #source: bti-far-3a.s
 #source: bti-far-3b.s
 #source: bti-far-3c.s
-#as: -mabi=lp64
 #ld: -shared -T bti-far-3.ld
 #objdump: -dr
 
diff --git a/ld/testsuite/ld-aarch64/bti-far-opt.d b/ld/testsuite/ld-aarch64/bti-far-opt.d
index ff20d0c2825..d06532da2d2 100644
--- a/ld/testsuite/ld-aarch64/bti-far-opt.d
+++ b/ld/testsuite/ld-aarch64/bti-far-opt.d
@@ -1,7 +1,6 @@
 #name: Check linker stubs with indirect calls handle BTI when target has BTI.
 #source: bti-far-opt.s
 #target: [check_shared_lib_support]
-#as: -mabi=lp64
 #ld: -shared -T bti-far.ld
 #objdump: -dr
 
-- 
2.43.0

